pkgs/top-level/all-packages.nix

```diff
@@ -915,30 +915,22 @@ with pkgs;
 
   csharp-ls = callPackage ../development/tools/language-servers/csharp-ls { };
 
-  dotnetCorePackages = recurseIntoAttrs (callPackage ../development/compilers/dotnet {});
+  dotnetPackagesFor-bin = callPackage ../development/compilers/dotnet {};
 
-  dotnet-sdk_6 = dotnetCorePackages.sdk_6_0;
-  dotnet-sdk_7 = dotnetCorePackages.sdk_7_0;
-  dotnet-sdk_8 = dotnetCorePackages.sdk_8_0;
+  dotnetPackages_6-bin = recurseIntoAttrs (dotnetPackagesFor-bin ../development/compilers/dotnet/versions/6.0.nix);
+  dotnetPackages_7-bin = recurseIntoAttrs (dotnetPackagesFor-bin ../development/compilers/dotnet/versions/7.0.nix);
+  dotnetPackages_8-bin = recurseIntoAttrs (dotnetPackagesFor-bin ../development/compilers/dotnet/versions/8.0.nix);
 
-  dotnet-runtime_6 = dotnetCorePackages.runtime_6_0;
-  dotnet-runtime_7 = dotnetCorePackages.runtime_7_0;
-  dotnet-runtime_8 = dotnetCorePackages.runtime_8_0;
+  dotnetPackages_6 = dotnetPackages_6-bin;
+  dotnetPackages_7 = dotnetPackages_7-bin;
+  dotnetPackages_8 = dotnetPackages_8-bin;
 
-  dotnet-aspnetcore_6 = dotnetCorePackages.aspnetcore_6_0;
-  dotnet-aspnetcore_7 = dotnetCorePackages.aspnetcore_7_0;
-  dotnet-aspnetcore_8 = dotnetCorePackages.aspnetcore_8_0;
-
-  dotnet-sdk = dotnetCorePackages.sdk_6_0;
-  dotnet-runtime = dotnetCorePackages.runtime_6_0;
-  dotnet-aspnetcore = dotnetCorePackages.aspnetcore_6_0;
-
-  buildDotnetModule = callPackage ../build-support/dotnet/build-dotnet-module { };
-  nuget-to-nix = callPackage ../build-support/dotnet/nuget-to-nix { };
+  buildDotnetModule = callPackage ../build-support/dotnet/build-dotnet-module { dotnetPackages = dotnetPackages_6; };
+  nuget-to-nix = callPackage ../build-support/dotnet/nuget-to-nix { dotnetPackages = dotnetPackages_6; };
```

Comment:
One question, I don't have strong feelings about this in either direction: What is the use case for having a top-level `buildDotnetModule`, `nuget-to-nix`, and `buildDotnetGlobalTool`?

Reply:
I don't think there's a reason to keep them.  I can move them to aliases.

Reply:
I believe it'd be better to just outright move then to aliases as a warning.